common/url.go

```diff
@@ -711,8 +711,16 @@ func MergeURL(serviceURL *URL, referenceURL *URL) *URL {
 
 	// finally execute methodConfigMergeFcn
 	for _, method := range referenceURL.Methods {
-		for _, fcn := range methodConfigMergeFcn {
-			fcn("methods." + method)
+		for _, paramKey := range []string{constant.LoadbalanceKey, constant.ClusterKey, constant.RetriesKey, constant.TimeoutKey} {
+			if v := referenceURL.GetParam(paramKey, ""); len(v) > 0 {
+				params[paramKey] = []string{v}
+			}
+
+			if len(mergedURL.GetParam("methods."+method+"."+paramKey, "")) == 0 {
```

Comment:
Please consider creating a variable to avoid concatenating '"methods."+method+"."+paramKey' many times, like 'key := "methods."+method+"."+paramKey'.